velox/vector/ComplexVector.cpp

```diff
@@ -1781,4 +1782,266 @@ void RowVector::appendNulls(vector_size_t numberOfRows) {
   bits::fillBits(mutableRawNulls(), oldSize, newSize, bits::kNull);
 }
 
+namespace {
+template <typename KeyType, TypeKind ValueKind>
```

Comment:
It looks like ValueKind template parameter is not needed. It is only used to check if replaceNulls is supported or not and that check can happen before calling this function.